### ISSUE TITLE
add explanation 

### ISSUE THREAD

**Issue by jonschlinkert** [doowb/githubbot#1]
would be useful to a reason this is useful to the GitHub project description (above the repo) and to the readme, so users know what this can be used for.

awesome btw!

**Comment by doowb** [comment]
I hadn't finished and wasn't sure how to explain the purpose of this module, but I think I figured it out. I updated the description and added a little extra explanation in the main readme part.

I think once this gets used a little more, I might add some "first-class" models around the objects that are found in payloads. Either add here or it might be something good to add to `github-base` since a lot of those objects are going to be shared with other parts of the github-api.

I was also thinking that "actions" could be added to the bot that would basically be wrapper methods around `github-base` for interacting with the github api.

**Comment by jonschlinkert** [comment]
> "actions" could be added to the bot

Good idea, maybe those could be like `presets`? Or plugins?

**Comment by jonschlinkert** [comment]
Oh, for the explanation I was thinking more along the lines of "create your own bot for doing things like automatically responding to issues"

edit: an actual use case I mean. like what would a developer do with the payloads and webhooks?

**Comment by doowb** [comment]
yeah... I was thinking that a plugin would add the "issueComment" action (which is just a method). Then a user would do something like:

``` js
bot.onIssues(function(payload, cb) {
  var repo = payload.repo;
  this.issueComment(repo, 'sweet!!!', function(err, results) {
    if (err) return cb(err);
    cb(null, payload);
  });
});
```

Either way... it allows plugins to be created independently of this library (like we've been doing) and users can choose how they're implement their bot.

**Comment by doowb** [comment]
> an actual use case I mean.

K.

**Comment by jonschlinkert** [comment]
> I was thinking that a plugin would add the "issueComment" action (which is just a method).

that's awesome. I like how this was done. this lib will be extremely useful - I would think for any big project
